### PR TITLE
create egress perimeter controls page

### PR DESCRIPTION
**Pull request by MichaelTrestman**
Previews:
[cockroachcloud/egress-perimeter-controls.md](https://deploy-preview-15426--cockroachdb-docs.netlify.app/docs/cockroachcloud/egress-perimeter-controls.html)
[cockroachcloud/network-authorization.md](https://deploy-preview-15426--cockroachdb-docs.netlify.app/docs/cockroachcloud/network-authorization.html)
[cockroachcloud/private-clusters.md](https://deploy-preview-15426--cockroachdb-docs.netlify.app/docs/cockroachcloud/private-clusters.html)
[cockroachcloud/security-overview.md](https://deploy-preview-15426--cockroachdb-docs.netlify.app/docs/cockroachcloud/security-overview.html)

### CODE REVIEW

**Comment by MichaelTrestman** [comment]
https://cockroachlabs.atlassian.net/browse/DOC-4831

File: cockroachcloud/egress-perimeter-controls.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+      "destination": "123.34.62.123/32",
+      "ports": [
+        443
+      ],
+      "paths": [],
+      "description": "egress for Kafka"
+    }
+    ~~~
+
+1. Make the POST request to create each new rule.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:230]:
```suggestion
1. Make a `POST` request to create each rule from the JSON payload.
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules" \
+    --data "@egress-rule2"
+    ~~~
+
+    ~~~json
+
+    ~~~
+
+    {{site.data.alerts.callout_danger}}
+    Your cluster's firewall behavior is not updated instantly when you submit the API request. After, submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:252]:
```suggestion
    Your cluster's firewall behavior is not updated instantly when you submit the API request. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```
Any idea how long you might need to wait?

File: cockroachcloud/egress-perimeter-controls.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+    {{site.data.alerts.end}}
+
+### Check the status of a rule
+
+{{site.data.alerts.callout_info}}
+Consult the glossary of [rule statuses](#rule-statuses).
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:258]:
```suggestion
Refer to the list of [rule statuses](#rule-statuses).
```
I'd update the language in other sections to match this. it's not strictly a glossary.

File: cockroachcloud/egress-perimeter-controls.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+{{site.data.alerts.end}}
+
+[API spec]
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:261]: Are these TODO?

File: cockroachcloud/egress-perimeter-controls.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+curl --request GET \
+  --url 'https://cockroachlabs.cloud/api/v1/clusters/$CLUSTER_ID/networking/egress-rules' \
+  --header "Authorization: Bearer $CC_API_KEY"
+~~~
+
+~~~txt
+
+~~~
+
+### Removing a rule
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:293]:
```suggestion
### Remove a rule
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+
+A `DELETE` request to the rule's path tells the API to delete the rule.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:295]:
```suggestion
To delete a rule, make `DELETE` request to the rule's path.
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+
+{{site.data.alerts.callout_danger}}
+Your cluster's firewall behavior is not updated instantly when you submit the API request. After, submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the deletion is complete.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:298]: Reminder to keep this in sync with other places the same text is used.

File: cockroachcloud/egress-perimeter-controls.md [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+
+{% include_cached copy-clipboard.html %}
+~~~shell
+curl --request DELETE \
+  --url 'https://cockroachlabs.cloud/api/v1/clusters/$CLUSTER_ID/networking/egress-rules/$RULE_ID' \
+  --header "Authorization: Bearer $CC_API_KEY"
+~~~
+
+~~~txt
+API RESPONSE!
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:309]: TODO?

File: cockroachcloud/egress-perimeter-controls.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+~~~
+
+
+## Rule statuses
+
+The API display the following statuses for a rule, when you [inspect an egress rule] or [list a cluster's egress rules]:
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:315]: TODO finish the links

File: cockroachcloud/egress-perimeter-controls.md [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+- `PENDING_DELETION`: The rule-removal is being implemented. The behavior of the cluster's network firewall may still reflect the target rule, so egress may still be allowed to the destination.
+- `DELETION_FAILED`: A rule update or a rule removal has failed. The behavior of the cluster's network firewall may still reflect the previous state of the rule. This is a rare occurrence and does indicate that you should [contact the Cockroach Labs Support Team](https://support.cockroachlabs.com) immediately, rather than attempting to retry rule creation, to avoid leaving your cluster's firewall in an unknown state.
+- `INCONSISTENT`: This indicates that a rule cannot apply consistently across all of the regions or compute resources to which it is meant to apply. This is a rare occurrence and does indicate that you should [contact the Cockroach Labs Support Team](https://support.cockroachlabs.com) immediately, rather than attempting to retry rule creation, to avoid leaving your cluster's firewall in an unknown state.
+
+
+## Troubleshooting
+
+The process
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:328]: Weird formatting?

File: cockroachcloud/network-authorization.md [hunk 11]
```diff
@@ -7,6 +7,10 @@ docs_area: manage
 
 To prevent denial-of-service and brute force password attacks, {{ site.data.products.dedicated }} requires you to authorize networks that can access your cluster by [allowlisting the public IP addresses](#ip-allowlisting) for your application. Optionally, you can [set up Virtual Private Cloud (VPC) peering](#vpc-peering) or [AWS PrivateLink](#aws-privatelink) for your cluster for enhanced network security and lower network latency.
 
+{{site.data.alerts.callout_success}}
+For additional cluster security, you can learn more about [Private Clusters (Preview)](private-clusters.html). A private cluster's nodes have no public IP addresses.
```
Comment (mdlinville) [inline cockroachcloud/network-authorization.md:11]:
```suggestion
[Private Clusters (Preview)](private-clusters.html), whose nodes have no public IP addresses, provide additional cluster security.
```

File: cockroachcloud/private-clusters.md [hunk 12]
```diff
@@ -0,0 +1,46 @@
+---
```
Comment (mdlinville) [inline cockroachcloud/private-clusters.md:1]: Noting that this topic actually comes from https://github.com/cockroachdb/docs/pull/15342

File: cockroachcloud/egress-perimeter-controls.md [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+Operators of {{ site.data.products.dedicated }} clusters can remove this risk by using Egress Perimiter Controls. This feature enables [admins](../{{site.versions["stable"]}}/security-reference/authorization.html#admin-role) to restrict egress to a specified external destinations. This adds a strong layer of protection against malicious data exfiltration. Along with other measures such as [Private Clusters](private-clusters.html), this is an important component in an overall strategy for maximizing network security.
+
+Further reading: [review how CockroachDB products differs in advanced security features](../{{site.versions["stable"]}}/security-reference/security-overview.html).
+
+
+{{site.data.alerts.callout_info}}
+Regardless of user-specific Egress Perimiter Control policy, egress is allowed to services managed by Cockroach Labs and essential to your cluster's functionality.
+{{site.data.alerts.end}}
+
+The Cloud API provides a real-time status for if the rule enablement is in progress or is complete.
```
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:33]:
```suggestion
The [Cloud API](cloud-api.html) provides a real-time status for if the rule enablement is in progress or is complete.
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+
+## Using Egress Perimeter Controls with the Cloud Console API
```
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:35]: That's true for steps, but isn't a step, it's a whole context, more like a page title, where we do typically use gerunds, "Managing..." So I disagree on this one.

File: cockroachcloud/egress-perimeter-controls.md [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+    ~~~
+
+    ~~~json
+    {
+      "id": "f0d06734-55ba-42e2-b606-2c1f58e6b9eb",
+      "name": "able-quetzal",
+      "cockroach_version": "v22.1.8",
+      "plan": "DEDICATED",
+      "cloud_provider": "AWS",
+      "account_id": "000906112402",
```
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:81]:
```suggestion
      "account_id": "1234567",
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 16]
```diff
@@ -0,0 +1,335 @@
...
+    --header "Authorization: Bearer $CC_API_KEY" \
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-traffic-policy" \
+    --data '{"allow_all":false}'
+    ~~~
+
+    ~~~txt
+    ???
+    ~~~
+
+1. Check your cluster's status and confirm the update is complete before proceeding.
```
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:130]: I don't know, I can't access the API. @carloruiz how do you tell if the update is complete?

File: cockroachcloud/egress-perimeter-controls.md [hunk 17]
```diff
@@ -0,0 +1,335 @@
...
+    curl --request GET \
+      --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID" \
+      --header "Authorization: Bearer $CC_API_KEY"
+    ~~~
+
+### Create egress rules to allowed destinations 
+
+An egress rule is created with the following attributes:
+
+- `name`: A name for the rule
```
Comment (MichaelTrestman) [inline cockroachcloud/egress-perimeter-controls.md:143]:
```suggestion
- `name`: A name for the rule.
```

File: cockroachcloud/egress-perimeter-controls.md (hunk 10 above)
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md]: whoops

File: cockroachcloud/egress-perimeter-controls.md [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+
+The [Cloud API](cloud-api.html) provides a real-time status for if the rule enablement is in progress or is complete.
+
+## Using Egress Perimeter Controls with the Cloud Console API
+
+### Prerequisites
+
+- You need a Cockroach Cloud account with billing enabled. Egress Perimeter Controls are available only for {{ site.data.products.dedicated }}, not {{ site.data.products.serverless }} clusters, and {{ site.data.products.dedicated }} clusters cost money to run.
+
+- Egress Perimeter Controls require your cluster to be a **Private Cluster**, with no public IP addresses on its nodes. Refer to [Private Clusters](private-clusters.html).
```
Comment (carloruiz) [inline cockroachcloud/egress-perimeter-controls.md:38]: This is actually not strictly required at an infrastructure level. But Maybe @Abhinav wants to make it a prerequisite from a business perspective

File: cockroachcloud/egress-perimeter-controls.md [hunk 19]
```diff
@@ -0,0 +1,327 @@
...
+
+- An API key with permissions to `edit` clusters in your org is required. You can provision API keys in the cloud console: [Create API Keys](console-access-management.html#create-api-keys)
```
Comment (carloruiz) [inline cockroachcloud/egress-perimeter-controls.md:40]: Only `admin` role can write to the egress API.

File: cockroachcloud/egress-perimeter-controls.md [hunk 20]
```diff
@@ -0,0 +1,335 @@
...
+      "updated_at": "2022-10-25T17:01:27.566551Z",
+      "deleted_at": null
+    }
+    ~~~
+
+
+### Change your cluster's egress traffic policy to deny-by-default
+
+{{site.data.alerts.callout_info}}
+External traffic destined to CRL-managed resources will always be allowed regardless of user-specified egress policy.
```
[the remark this answers is not in the document]
Comment (carloruiz) [reply cockroachcloud/egress-perimeter-controls.md:110]: Ah the EgressRule response has a field called `crl_managed`, which leaks the crl name. Too late to change now :/

File: cockroachcloud/egress-perimeter-controls.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (carloruiz) [reply cockroachcloud/egress-perimeter-controls.md:127]: Hmm, there's a field in the `intrusion_crdb_clusters` table called `restricted_egress`. If Set to true, then the update has been successful. I know @rgcase was trying to expose this field in the console response for get cluster. 

This is an oversight on my end as without it, there's no way users can tell if the default-deny policy has taken effect :grimace.

File: cockroachcloud/egress-perimeter-controls.md [hunk 21]
```diff
@@ -0,0 +1,327 @@
...
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules" \
+    --data "@egress-rule2"
+    ~~~
+
+    ~~~json
+
+    ~~~
+
+    {{site.data.alerts.callout_danger}}
+    Your cluster's firewall behavior is not updated instantly when you submit the API request. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```
Comment (carloruiz) [inline cockroachcloud/egress-perimeter-controls.md:249]: I'd reword this to `s/Your cluster's firewall behavior is not updated instantly when you submit the API request/Your cluster's firewall behavior is enforced asynchronously after the API response.`

File: cockroachcloud/egress-perimeter-controls.md (hunk 8 above)
Comment (carloruiz) [reply cockroachcloud/egress-perimeter-controls.md:306]: A full `EgressRule` is returned in the delete response.

File: cockroachcloud/egress-perimeter-controls.md (hunk 9 above)
Comment (carloruiz) [reply cockroachcloud/egress-perimeter-controls.md:312]: `s/display/displays`

File: cockroachcloud/egress-perimeter-controls.md [hunk 22]
```diff
@@ -0,0 +1,327 @@
...
+Regardless of user-specific Egress Perimiter Control policy, egress is always permitted to services that are managed by Cockroach Labs and are essential to your cluster's functionality.
+{{site.data.alerts.end}}
+
+The [Cloud API](cloud-api.html) provides a real-time status for if the rule enablement is in progress or is complete.
+
+## Using Egress Perimeter Controls with the Cloud Console API
+
+### Prerequisites
+
+- You need a Cockroach Cloud account with billing enabled. Egress Perimeter Controls are available only for {{ site.data.products.dedicated }}, not {{ site.data.products.serverless }} clusters, and {{ site.data.products.dedicated }} clusters cost money to run.
```
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:36]: Ha yeah, I forgot about this little thing... I'm just going to delete the note. We don't have warnings like this in general. I was wondering about how to message the fact that like, you can't do this 'tutorial' without having to pay for it... but I think it's unnecessary since it's since been decided to make it private preview.
Thanks for catching it!

File: cockroachcloud/egress-perimeter-controls.md (hunk 18 above)
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:38]: I believe it is being declared as a requirement despite not strictly being, that's what we decided at the meeting yesterday, correct @abhinavg6 ?

File: cockroachcloud/egress-perimeter-controls.md (hunk 19 above)
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:40]: cool, thanks

File: cockroachcloud/egress-perimeter-controls.md (hunk 20 above)
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:110]: Good point Carlo, but I think Matt's point still stands.

File: cockroachcloud/egress-perimeter-controls.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:127]: Hmm. To understand this, there is currently no way to verify if Egress Perimeter Controls has been enabled or not on the API?

File: cockroachcloud/egress-perimeter-controls.md [hunk 23]
```diff
@@ -0,0 +1,324 @@
...
+    --url "https://management-staging.crdb.io/api/v1/clusters/$CLUSTER_ID/networking/egress-rules" \
+    --data "@egress-rule2"
+    ~~~
+
+    ~~~json
+
+    ~~~
+
+    {{site.data.alerts.callout_danger}}
+    Your cluster's firewall behavior is not updated instantly when you submit the API request. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```
Comment (MichaelTrestman) [inline cockroachcloud/egress-perimeter-controls.md:249]:
```suggestion
    Your cluster's firewall behavior is enforced asynchronously after the API response. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the new rules have been created.
```

File: cockroachcloud/egress-perimeter-controls.md (hunk 21 above)
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:249]: thx, much better!

File: cockroachcloud/egress-perimeter-controls.md (hunk 4 above)
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:258]: yes

File: cockroachcloud/egress-perimeter-controls.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (MichaelTrestman) [reply cockroachcloud/egress-perimeter-controls.md:127]: Essentially, you can check if Egress Perimeter Controls have been enabled by GETting "clusters/$CLUSTER_ID/networking/egress-rules", which will... I guess error if the feature isn't enabled? I think I'll suggest that, after verifying it once I have API access.

[the remark this answers is not in the document]
Comment (carloruiz) [reply cockroachcloud/egress-perimeter-controls.md:127]: Nope, the functionality still works even though restricted egress is not set. The difference is the egress rules are not enforced. I'll put up a PR that surfaces the `egress traffic policy` field on the get cluster response. It won't make it to prod until next week though.

File: cockroachcloud/egress-perimeter-controls.md [hunk 24]
```diff
@@ -19,40 +19,37 @@ This page describes the reasons to use Egress Perimeter Controls for enhanced se
...
+Operators of {{ site.data.products.dedicated }} clusters can mitigate against this risk by using Egress Perimeter Controls, which enable [admins](../{{site.versions["stable"]}}/security-reference/authorization.html#admin-role) to restrict egress to a list of specified external destinations. This adds a strong layer of protection against malicious data exfiltration. Along with other measures such as [Private Clusters](private-clusters.html), Egress Perimeter Controls are an important component in an overall strategy for maximizing network security.
 
 Further reading: [review how CockroachDB products differs in advanced security features](../{{site.versions["stable"]}}/security-reference/security-overview.html).
 
 
 {{site.data.alerts.callout_info}}
-Regardless of user-specific Egress Perimiter Control policy, egress is allowed to services managed by Cockroach Labs and essential to your cluster's functionality.
+Regardless of user-specific Egress Perimiter Control policy, egress is always permitted to services that are managed by Cockroach Labs and are essential to your cluster's functionality.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:30]: Do a scrub for "Perimiter". 👼

File: cockroachcloud/egress-perimeter-controls.md [hunk 25]
```diff
@@ -19,40 +19,37 @@ This page describes the reasons to use Egress Perimeter Controls for enhanced se
...
 {{site.data.alerts.end}}
 
-The Cloud API provides a real-time status for if the rule enablement is in progress or is complete.
 
-## Using Egress Perimeter Controls with the Cloud Console API
+## Prerequisites to using Egress Perimeter Controls with the Cloud Console API
 
-### Prerequisites
+- You need a Cockroach Cloud account with billing enabled. Egress Perimeter Controls are available only for {{ site.data.products.dedicated }}, not {{ site.data.products.serverless }} clusters, and {{ site.data.products.dedicated }} have a non-zero operating cost that must be billed.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:36]: Use the variable for Cockroach Cloud wherever possible.

File: cockroachcloud/egress-perimeter-controls.md [hunk 26]
```diff
@@ -19,40 +19,37 @@ This page describes the reasons to use Egress Perimeter Controls for enhanced se
...
 
-- You will need a Cockroach Cloud account with billing enabled, as Egress Perimeter Controls are only available for {{ site.data.products.dedicated }}, not {{ site.data.products.serverless }} clusters, and {{ site.data.products.dedicated }} clusters cost money to run.
+- Egress Perimeter Controls require your cluster to be a **Private Cluster**, with no public IP addresses on its nodes. Refer to [Private Clusters](private-clusters.html).
 
-- Egress Perimeter Controls are currently only enabled for [Private Clusters](private-clusters.html).
-
-- You will need an API key with permissions to `edit` clusters in your org. You can provision API keys in the cloud console: [Create API Keys](console-access-management.html#create-api-keys)
+- An API key with `admin` privilege on clusters in your org is required. You can provision API keys in the cloud console: [Create API Keys](console-access-management.html#create-api-keys)
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:40]: It might be overkill, but perhaps a link to admin role.

File: cockroachcloud/egress-perimeter-controls.md [hunk 27]
```diff
@@ -19,40 +19,37 @@ This page describes the reasons to use Egress Perimeter Controls for enhanced se
...
-### Initialize your shell with your API key and Cluster id
+## Initialize your shell with your API key and Cluster id
 
 1. Inspect your cluster in the [clusters page](https://cockroachlabs.cloud/clusters) in the {{ site.data.products.db }} console.
 
-1. Find your cluster's universally unique identifier (UUID). To do this, click on the name of your cluster from the [clusters page](https://cockroachlabs.cloud/clusters) in the console. The UUID will appear in the URL of the overview page for that specific cluster, in the format:
+1. Find your cluster's universally unique identifier (UUID). To do this, select your cluster from the [clusters page](https://cockroachlabs.cloud/clusters) in the console. The UUID will appear in the URL of the overview page for that specific cluster, in the format:
 
-`https://cockroachlabs.cloud/cluster/{ your cluster's UUID }/overview`
+    `https://cockroachlabs.cloud/cluster/{ your cluster's UUID }/overview`
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:52]: I think we tend to use underscores and lowercase for placeholders.

File: cockroachcloud/egress-perimeter-controls.md [hunk 28]
```diff
@@ -129,28 +126,28 @@ External traffic destined to CRL-managed resources will always be allowed regard
...
+- `name`: A name for the rule.
+- `type`: Whether the destination will be specified as a fully-qualified domain name (FQDN), or as a range of IP addresses specified in CIDR notation. Value is `"FQDN"` or `"CIDR"`.
 - `destination`: Either a fully qualified domain name, for example `www.cockroachlabs.com`, or a CIDR range, for example, `123.45.67.890/32`.
-- `ports`: An array of allowed ports, for example [44,8080]
+- `ports`: An array of allowed ports, for example `[44,8080]`.
     {{site.data.alerts.callout_danger}}
     By default, all ports are allowed.
     {{site.data.alerts.end}}
 - `paths`: An array of allowed destination paths.
 - `description`:
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:148]: Need a description for the description.

File: cockroachcloud/egress-perimeter-controls.md (hunk 7 above)
Comment (mdlinville) [reply cockroachcloud/egress-perimeter-controls.md:295]: Re-reminder since you changed the other instance based on feedback. 👼

File: cockroachcloud/egress-perimeter-controls.md [hunk 29]
```diff
@@ -0,0 +1,440 @@
...
+  "pagination": null
+}
+~~~
+
+## Remove a rule
+
+To delete a rule, make `DELETE` request to the rule's path.
+
+{{site.data.alerts.callout_danger}}
+Your cluster's firewall behavior is enforced asynchronously after the API response. After, submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the deletion is complete.
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:395]:
```suggestion
Your cluster's firewall behavior is enforced asynchronously after the API response. After submitting the request, [check your egress rules](#check-a-clusters-egress-rules-allowed-destinations) to confirm that the deletion is complete.
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 30]
```diff
@@ -0,0 +1,324 @@
...
+~~~
+
+~~~txt
+API RESPONSE!
+~~~
+
+
+## Rule statuses
+
+The API displays the following statuses for a rule, when you [inspect an egress rule] or [list a cluster's egress rules]:
```
Comment (abhinavg6) [inline cockroachcloud/egress-perimeter-controls.md:428]: This is an awesome list and very well documented. Thank you.

**Review by abhinavg6 — CHANGES_REQUESTED** [review]

File: cockroachcloud/egress-perimeter-controls.md [hunk 31]
```diff
@@ -0,0 +1,324 @@
...
+
+Operators of {{ site.data.products.dedicated }} clusters can mitigate against this risk by using Egress Perimeter Controls, which enable [admins](../{{site.versions["stable"]}}/security-reference/authorization.html#admin-role) to restrict egress to a list of specified external destinations. This adds a strong layer of protection against malicious data exfiltration. Along with other measures such as [Private Clusters](private-clusters.html), Egress Perimeter Controls are an important component in an overall strategy for maximizing network security.
+
+Further reading: [review how CockroachDB products differs in advanced security features](../{{site.versions["stable"]}}/security-reference/security-overview.html).
+
+
+{{site.data.alerts.callout_info}}
+Regardless of user-specific Egress Perimiter Control policy, egress is always permitted to services that are managed by Cockroach Labs and are essential to your cluster's functionality.
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply cockroachcloud/egress-perimeter-controls.md:30]: @abhinavg6 updated your suggestion to "policies". Is that correct?

[the remark this answers is not in the document]
Comment (abhinavg6) [reply cockroachcloud/egress-perimeter-controls.md:30]: Actually ''policy" is relevant Matt, because a policy is composed of multiple egress rules. The plural may confuse people that there are multiple policies across egress rules.

File: cockroachcloud/egress-perimeter-controls.md [hunk 32]
```diff
@@ -0,0 +1,440 @@
...
+An egress rule is created with the following attributes:
+
+- `name`: A name for the rule.
+- `type`: Whether the destination will be specified as a fully-qualified domain name (FQDN), or as a range of IP addresses specified in CIDR notation. Value is `"FQDN"` or `"CIDR"`.
+- `destination`: Either a fully qualified domain name, for example `www.cockroachlabs.com`, or a CIDR range, for example, `123.45.67.890/32`.
+- `ports`: An array of allowed ports, for example `[44,8080]`.
+    {{site.data.alerts.callout_danger}}
+    By default, all ports are allowed.
+    {{site.data.alerts.end}}
+- `paths`: An array of allowed destination paths.
```
Comment (abhinavg6) [inline cockroachcloud/egress-perimeter-controls.md:151]: Would be good to provide an example for this too.

File: cockroachcloud/egress-perimeter-controls.md [hunk 33]
```diff
@@ -0,0 +1,440 @@
...
+    type: "FQDN"
+    destination: "storage.googleapis.com"
+    ports: [443, 80]
+    paths:
+        - "/customer-manged-bucket-1/*" # replace with bucket path
+        - "/customer-manged-bucket-2/*" # replace with bucket path
+    description: 'egress for GCP storage buckets'
+    ~~~
+
+1. Next, create a YAML manifest for a CIDR-based rule.
```
Comment (abhinavg6) [inline cockroachcloud/egress-perimeter-controls.md:171]:
```suggestion
1. Next, create a YAML manifest for a CIDR-based rule. This example adds egress to a self-managed Kafka cluster.
```

File: cockroachcloud/egress-perimeter-controls.md [hunk 34]
```diff
@@ -0,0 +1,440 @@
...
+- `ports`: An array of allowed ports, for example `[44,8080]`.
+    {{site.data.alerts.callout_danger}}
+    By default, all ports are allowed.
+    {{site.data.alerts.end}}
+- `paths`: An array of allowed destination paths.
+- `description`:
+
+The following steps create one FQDN rule and one CIDR rule.
+
+1. First, create a YAML manifest for a FQDN-based rule. This example, adds egress to a Google Cloud Platform (GCP) storage bucket.
```
Comment (abhinavg6) [inline cockroachcloud/egress-perimeter-controls.md:156]:
```suggestion
1. First, create a YAML manifest for a FQDN-based rule. This example adds egress to a couple of Google Cloud Platform (GCP) storage buckets.
```

**Review by abhinavg6 — CHANGES_REQUESTED** [review]
Some more comments and responded to a question from Matt.

**Comment by carloruiz** [comment]
LGTM from a technical standpoint. Also the words flow very smoothly, making the doc a nice read 👍

**Comment by mdlinville** [comment]
@MichaelTrestman merged `master` after I merged #15342

File: cockroachcloud/egress-perimeter-controls.md [hunk 35]
```diff
@@ -64,18 +62,19 @@ The operations described in this page require an API key with very broad permiss
...
 
     ~~~json
+    
     {
-      "id": "f0d06734-55ba-42e2-b606-2c1f58e6b9eb",
-      "name": "able-quetzal",
-      "cockroach_version": "v22.1.8",
+      "id": "21f474d5-3e65-4a54-9317-e8d8803ef917",
+      "name": "docstest",
+      "cockroach_version": "latest-v22.2-build(sha256:e42c4de8577556132120a9ab07efc1a2a96779c028ebab99223d862d9792428b)",
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:74]: Probably just a nit, but I wonder if this should be changed to the 22.1.9 build hash. Unless this feature requires v22.x.

File: cockroachcloud/egress-perimeter-controls.md [hunk 36]
```diff
@@ -0,0 +1,418 @@
...
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule1.yml > egress-rule1.json
+    ruby -ryaml -rjson -e 'puts(YAML.load(ARGF.read).to_json)' < egress-rule2.yml > egress-rule2.json
+    ~~~
+
+1. Use the shell utility `jq` to inspect the JSON payload:
+    
+    {{site.data.alerts.callout_info}}
+    On macOS, you can install `jq` from Homebrew: `brew install jq`
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:181]:
```suggestion
    On macOS, you can install `jq` from Homebrew: 
    
    {% include_cached copy-clipboard.html %}
    ~~~shell
    `brew install jq`
    ~~~
```
Very optional: I suspect this info is more usable if it's a code block, but I'll leave the decision entirely up to you.

File: cockroachcloud/egress-perimeter-controls.md [hunk 37]
```diff
@@ -0,0 +1,418 @@
...
+    "ports": [
+      443
+    ],
+    "paths": [],
+    "description": "egress for Kafka",
+    "created_at": "2022-10-27T19:36:25.670162Z"
+  }
+~~~
+
+## Check a cluster's egress rules/allowed destinations
```
Comment (mdlinville) [inline cockroachcloud/egress-perimeter-controls.md:310]:
```suggestion
## Check a cluster's egress rules and allowed destinations
```

**Review by mdlinville — APPROVED** [review]
LGTM, really easy to understand. I left a few minor cosmetic suggestions when reviewing the HTML preview.
